### DISCUSSION TITLE
Need direction: Learning Zephyr with ESP32

### DISCUSSION THREAD

**Discussion by ams-tech13** [zephyrproject-rtos/zephyr#48761] [General]
Hello,

Can anyone point me to good resources, keywords etc for the below described application?

My plan is to connect an ESP32 to the network and just insert randomly generated value to a database like influxDB or even ElephantSQL, it does not matter which one, you can suggest something else as well.

So far I have been succesfully able to run the sample codes, and connect to the wifi, but I am still somewhat lost. Been reading, but its too much. 

I have done this previously with arduino, raspberry pi etc. But i want to get my hands on Zephyr now. 

Any help is greatly appreciated. Thank you

**Comment by rftafas** [comment]
I think after the tutorial on how to bring up Zephyr to your ESP32, most of the work is more Zephyr than actually ESP32.

The bring up: https://zephyrproject.org/zephyr-rtos-on-esp32/

EDIT: [I found this great and beginner friendly video.](https://youtu.be/Z_7y_4O7yTw)

**Reply by Dark1886** [reply]
Could you try and point to that video again? The second link just points back to this same GitHub discussion thread.

**Reply by rftafas** [reply]
Fixed. Tks!

**Comment by joelguittet** [comment]
Hello @ams-tech13 @Dark1886 @rftafas 

I'm reaching here because I m looking for information to properly setup the esp32 configuration. Unfortunately this is always blinking led or simple application so no real difficulty. I m blocked while trying to do more complex topics, communicating with https etc.

Do you have succeed with this ? @ams-tech13 you initially indicated you would like to work similar topics.

I opened https://github.com/zephyrproject-rtos/zephyr/discussions/62237 few days ago but no feedback at the moment.

Thanks for sharing !
Joel

**Comment by rftafas** [comment]
@joelguittet Did you check the latest videos on that channel I mentioned above? They even have a tutorial about writing a driver.

Other than that, [there is a tutorial in Portuguese on website Embarcados](https://embarcados.com.br/criando-uma-aplicacao-zephyr-out-of-tree-com-wifi-do-esp32-habilitado/) - google translator friendly - about out of tree application with WiFi. While it may be a bit outdated, it still should be useful.

**Reply by joelguittet** [reply]
Thanks for the quick reply!
I checked then and this link in Portuguese is not bad but all of this is ... trivial until now.
I'm searching since several days now an example running an HTTPS client, with just a simple GET or POST for example on a public server. DNS resolution + connect + send/recv data.
Goal is to properly configure heap, mbedtls etc.
If you have one i will be so happy :)
Joel

**Reply by jukkar** [reply]
> I'm searching since several days now an example running an HTTPS client, with just a simple GET or POST for example on a public server. DNS resolution + connect + send/recv data.

Note that there are networking samples for each of these functionality so
* `samples/net/sockets/http_client` for examples with GET/POST and mbedtls
* `samples/net/dns_resolve` for resolving IP addresses
* `samples/net/sockets/echo_client` for simple connect and send/recv data

There is no sample application that would do all of these in one app, but you can certainly propose one if needed.

**Reply by joelguittet** [reply]
Yep I know all these samples, the main issue is that they are not available with the esp32 board. Probably I will restart using the samples and propose a board overlay+conf if I succeed with this :-)

**Reply by jukkar** [reply]
Note that the HTTP/DNS/socket APIs should be vendor agnostics, so the those samples should work (with little conf + DT tweaking) with any board that can create a network interface.

**Reply by joelguittet** [reply]
I know that. To be very accurate and this is what I suppose in the other discussion I mentioned, my concern is the configuration of system heap size, mbedtls heap size, number of tx rx buffer etc. There are plenty of parameters. And topic is also how to fit everything in the esp32. Looks not trivial because of the different ram regions and their sizes.
All of these is not addressed, I haven't found an example to configure that properly.
This is why I said blinking led or hello world, or just connecting wifi network is not enough to demonstrate the esp32 can be used with zephyr in a real application. That I m trying to build :-)

**Comment by rftafas** [comment]
We don't have anything on that sense, unfortunately. Although this doesn't seem to be specific to ESP32, I do agree such example would be interesting.

The team from Golioth have something that could inspire you: https://blog.golioth.io/how-to-add-golioth-to-an-existing-zephyr-project/

Other than that, it is good & old documentation reading.

**Reply by joelguittet** [reply]
Looks a good starting point. Need to cleanup my project and properly read this :) Thanks for sharing
